crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -411,9 +411,16 @@ where
                 None => return Ok(None),
             };
 
-            this.build_transaction_receipt(tx, meta, receipt).await.map(Some)
+            Ok(Some((tx, meta, receipt)))
         })
-        .await
+        .await?;
+
+        let (tx, meta, receipt) = match result {
+            Some((tx, meta, receipt)) => (tx, meta, receipt),
+            None => return Ok(None),
+        };
+
+        self.build_transaction_receipt(tx, meta, receipt).await.map(Some)
```

Comment:
this is also okay since we can do the build_transaction work outside the blocking task